snapcraft/providers/_provider.py

```diff
@@ -133,6 +148,15 @@ def is_provider_available(cls) -> bool:
         :returns: True if installed.
         """
 
+    @abstractmethod
+    def environment(self, *, instance_name: str) -> Executor:
```

Comment:
```suggestion
    def create_environment(self, *, instance_name: str) -> Executor:
```

Reply:
That's my one comment for this PR, rest looks good.

Reply:
Renamed.